include/xtypes/idl/Module.hpp

```diff
@@ -65,6 +70,29 @@ class Module
         return inner_[submodule];
     }
 
+    using ModuleVisitor = std::function<void (const Module& mod)>;
+
+    void for_each_submodule(
+            ModuleVisitor visitor,
+            bool recursive = true) const
+    {
+        for (const auto& inner : this->inner_)
+        {
+            visitor(*inner.second.get());
+            if (recursive)
+            {
+                for_each_submodule(visitor, inner.second.get());
```

Comment:
Where is defined this `for_each_submodule` function? If it is calling itself, `inner.second.get()` is not a boolean, is it?

Reply:
This was wrong, fixed now